lib/Jsrt/JsrtDebuggerObject.cpp

```diff
@@ -0,0 +1,847 @@
+            resolvedObject.name = AnewNoThrowArray(this->debuggerObjectsManager->GetDebugObjectArena(), WCHAR, len + 1);
+            if (resolvedObject.name == nullptr)
+            {
+                return nullptr;
+            }
+            wcscpy_s((WCHAR*)resolvedObject.name, len + 1, source);
+
+            resolvedObject.typeId = Js::JavascriptOperators::GetTypeId(resolvedObject.obj);
+            JsrtDebuggerObjectBase::CreateDebuggerObject<JsrtDebuggerObjectProperty>(this->debuggerObjectsManager, resolvedObject, scriptContext, [&](Js::Var marshaledObj)
+            {
+                evalResult = (Js::DynamicObject*)marshaledObj;
+            });
+        }
+    }
+    return evalResult;
+}
+
+JsrtDebuggerObjectProperty::JsrtDebuggerObjectProperty(JsrtDebuggerObjectsManager* debuggerObjectsManager, WeakArenaReference<Js::IDiagObjectModelDisplay>* objectDisplay) :
+    JsrtDebuggerObjectBase(JsrtDebuggerObjectType::Property, debuggerObjectsManager),
+    objectDisplay(objectDisplay),
+    walkerRef(nullptr)
+{
+    Assert(objectDisplay != nullptr);
+}
+
+JsrtDebuggerObjectProperty::~JsrtDebuggerObjectProperty()
+{
+    if (this->objectDisplay != nullptr)
+    {
+        HeapDelete(this->objectDisplay);
+        this->objectDisplay = nullptr;
+    }
+
+    if (this->walkerRef != nullptr)
+    {
+        HeapDelete(this->walkerRef);
+        this->walkerRef = nullptr;
+    }
+}
+
+JsrtDebuggerObjectBase * JsrtDebuggerObjectProperty::Make(JsrtDebuggerObjectsManager* debuggerObjectsManager, WeakArenaReference<Js::IDiagObjectModelDisplay>* objectDisplay)
+{
+    JsrtDebuggerObjectBase* debuggerObject = Anew(debuggerObjectsManager->GetDebugObjectArena(), JsrtDebuggerObjectProperty, debuggerObjectsManager, objectDisplay);
+
+    debuggerObjectsManager->AddToDebuggerObjectsDictionary(debuggerObject);
+
+    return debuggerObject;
+}
+
+Js::DynamicObject * JsrtDebuggerObjectProperty::GetJSONObject(Js::ScriptContext* scriptContext)
+{
+    Js::IDiagObjectModelDisplay* objectDisplayRef = this->objectDisplay->GetStrongReference();
+
+    Js::DynamicObject* propertyObject = nullptr;
+
+    if (objectDisplayRef != nullptr)
+    {
+        propertyObject = scriptContext->GetLibrary()->CreateObject();
+
+        LPCWSTR name = objectDisplayRef->Name();
+
+        JsrtDebugUtils::AddPropertyToObject(propertyObject, JsrtDebugPropertyId::name, name, wcslen(name), scriptContext);
+
+        JsrtDebugUtils::AddPropertyType(propertyObject, objectDisplayRef, scriptContext); // Will add type, value, display, className, propertyAttributes
+
+        JsrtDebugUtils::AddPropertyToObject(propertyObject, JsrtDebugPropertyId::handle, this->GetHandle(), scriptContext);
+
+        this->objectDisplay->ReleaseStrongReference();
+    }
+
+    return propertyObject;
+}
+
+Js::DynamicObject* JsrtDebuggerObjectProperty::GetChildrens(Js::ScriptContext* scriptContext, uint fromCount, uint totalCount)
```

Comment:
done

---
In reply to: [62759434](https://github.com/Microsoft/ChakraCore/pull/926#discussion_r62759434) [](ancestors = 62759434)